### PR TITLE
test(Runtime): Staking Tests

### PR DESCRIPTION
**Pull request by kollegian**
Required for merge:
- [ ] `pr-workflow-check / draft-release-check` is ✅ success
- Other rules GitHub shows you, or can be read in [configuration](../terraform/github.com/branches.tf) 

Makes review faster:
- [X] PR title is my best effort to provide summary of changes and has clear text to be part of release notes 
- [X] I marked PR by `misc` label if it should not be in release notes
- [ ] Linked Zenhub/Github/Slack/etc reference if one exists
- [ ] I was clear on what type of deployment required to release my changes (node, runtime, contract, indexer, on chain operation, frontend, infrastructure) if any in PR title or description
- [ ] Added reviewer into `Reviewers`
- [ ] I tagged(`@`) or used other form of notification of one person who I think can handle best review of this PR
- [ ] I have proved that PR has no general regressions of relevant features and processes required to release into production
- [ ] Any dependency updates made, was done according guides from relevant dependency
- Clicking all checkboxes 
- Adding detailed description of changes when it feels appropriate (for example when PR is big)

### CODE REVIEW

**Comment by blasrodri** [comment]
@kkast @fl-y can you guys review?

**Comment by dzmitry-lahoda** [comment]
adding implementer pallets just in case @RustNinja  ;

**Comment by dzmitry-lahoda** [comment]
i am more interested in review of cross chain test.

File: code/integration-tests/runtime-tests/tests/utils/stakingUtils.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+) {
+  const poolCurrencyId = api.createType('u128', stakingTokenId);
+  const rewardCurrencyId = api.createType('u128', rewardTokenId);
+  const periodCount = api.createType('u32', period);
+  const amount = api.createType('u128', rewardAmount.toString());
+  return await sendAndWaitForSuccess(
+    api,
+    sudoKey,
+    api.events.farming.RewardScheduleUpdated.is,
+    api.tx.sudo.sudo(api.tx.farming.updateRewardSchedule(poolCurrencyId, rewardCurrencyId, periodCount, amount)),
```
Comment (RustNinja) [inline code/integration-tests/runtime-tests/tests/utils/stakingUtils.ts:59]: jfy @dzmitry-lahoda
in current version of pallet implementation it is not correct to update reward schedule if schedule started already and there is funds on pool account that user did not claimed yet.
here is the [pallet code](https://github.com/ComposableFi/composable/blob/main/code/parachain/frame/farming/src/lib.rs#L271-L272) that you need to be aware

so the actual schedule update supposed to be this steps
- remove current scheduled
- create the new one schedule
- transfer to the the pool not claimed reward from prev schedule 

In other case you should not call update reward schedule extrinsic.

to allow safely update existing reward pool on the fly we need to modify this [line of code](https://github.com/ComposableFi/composable/blob/main/code/parachain/frame/farming/src/lib.rs#L271-L272)

and recalculate next schedule reward based on current schedule that not finished yet.

**Review by RustNinja — APPROVED** [review]

**Pull request closed without merging**